### PR TITLE
[lpeg] Add cmake Config file

### LINKED ISSUE #38340: [luafilesystem] [lpeg] package config file
### Is your feature request related to a problem? Please describe.

Neither of these ports, indeed none of the lua libraries, currently provide package config files; so they are not easily found by cmake when they are installed.

I suspect normally these dlls are not linked and simply loaded at runtime through lua's require command, but that doesn't work when a static triplet is used.

### Proposed solution

Provide package config files for these ports e.g. lpegConfig.cmake

### Describe alternatives you've considered

Alternatively I've considered restricting these packages to always build with dlls/so files even when a static triplet is chosen. I think it would be more valuable to support linking the libraries as this differentiates vcpkg from luarocks and results in simpler packaging.

### Additional context

I am willing to work on the change myself if there is consensus.

### PR DESCRIPTION
**Pull request by TheCycoONE**
Fixes #38340 

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [ ] ~SHA512s are updated for each updated download.~
- [ ] ~The "supports" clause reflects platforms that may be fixed by this new version.~
- [ ] ~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~
- [ ] ~Any patches that are no longer applied are deleted from the port's directory.~
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/lpeg/portfile.cmake [hunk 1]
```diff
@@ -11,18 +11,23 @@ vcpkg_extract_source_archive(
 
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/CMakeLists.txt" DESTINATION "${SOURCE_PATH}")
 file(COPY "${CMAKE_CURRENT_LIST_DIR}/lpeg.def" DESTINATION "${SOURCE_PATH}")
+file(COPY "${CMAKE_CURRENT_LIST_DIR}/unofficial-lpeg-config.cmake.in" DESTINATION "${SOURCE_PATH}")
 
 vcpkg_cmake_configure(
     SOURCE_PATH "${SOURCE_PATH}"
+    OPTIONS
+        -DLPEG_VERSION="${VERSION}"
```
Comment (dg0yt) [inline ports/lpeg/portfile.cmake:19]:
```suggestion
        "-DLPEG_VERSION=${VERSION}"
```

**Review by dg0yt — CHANGES_REQUESTED** [review]

File: ports/lpeg/portfile.cmake (hunk 1 above)
Comment (MonicaLiu0311) [reply ports/lpeg/portfile.cmake:19]: Ping @TheCycoONE

**Comment by TheCycoONE** [comment]
Sure

**Comment by TheCycoONE** [comment]
Added license as well

**Review by MonicaLiu0311 — APPROVED** [review]

**Pull request merged**